src/components/spatial/Spatial.js

```diff
@@ -4,16 +4,89 @@ import {
   ScatterplotLayer, PolygonLayer, COORDINATE_SYSTEM, BitmapLayer, BaseTileLayer,
 } from 'deck.gl';
 import { load } from '@loaders.gl/core';
-import { SelectablePolygonLayer } from '../../layers';
+import { Texture2D } from '@luma.gl/webgl';
+import GL from '@luma.gl/constants';
+import { slice, openArray } from 'zarr';
+import { SelectablePolygonLayer, XRLayer } from '../../layers';
 import { tileToBoundingBox, getTileIndices } from './tiling-utils';
 import { cellLayerDefaultProps, PALETTE, DEFAULT_COLOR } from '../utils';
 import AbstractSelectableComponent from '../AbstractSelectableComponent';
 import LayersMenu from './LayersMenu';
 
+const zarrArrays = {};
+
 export function square(x, y, r) {
   return [[x, y + r], [x + r, y], [x, y - r], [x - r, y]];
 }
 
+async function getTexture(config, gl, tileSize, x, y, z, stride, width) {
+  const arrSlice = slice(stride * width * y + stride * x, stride * width * y + stride * (x + 1));
+  const zarrKey = config.zarrConfig.store + config.zarrConfig.path;
+  if (!zarrArrays[zarrKey]) {
+    zarrArrays[zarrKey] = await openArray(config.zarrConfig);
+  }
+  const arr = zarrArrays[zarrKey];
+  const dataSlice = await arr.get([arrSlice]);
+  const { data } = dataSlice;
+  // eslint-disable-next-line no-nested-ternary
+  const formats = data instanceof Uint8Array
+    ? {
+      format: GL.R8UI,
+      dataFormat: GL.RED_INTEGER,
+      type: GL.UNSIGNED_BYTE,
+    }
+    : (
+      data instanceof Uint16Array
+        ? {
+          format: GL.R16UI,
+          dataFormat: GL.RED_INTEGER,
+          type: GL.UNSIGNED_SHORT,
+        }
+        : {
+          format: GL.R32UI,
+          dataFormat: GL.RED_INTEGER,
+          type: GL.UNSIGNED_INT,
+        }
+    );
```

Comment:
This is in preparation of all data formats.  Please let me know if you can think of something better than the nested ternary.

Reply:
and this list could get longer over time? I see the repeated structure of `{format, dataFormat, type}`, and would like to bring that out. Maybe:
```javascript
const isInt8 = data instanceof Uint8Array;
// etc.
const formats = {
      format: isInt8 && GL.R8UI 
           || isInt16 && GL.R16UI 
           || isInt32 && GL.R32UI,
      dataFormat: GL.RED_INTEGER,
      type: isInt8 && GL.UNSIGNED_BYTE || ...,
    }

Reply:
The list will not get longer but that it very good :)